File: backend/src/main/java/com/tencheeduard/hotelapp/entities/Room.java
package com.tencheeduard.hotelapp.entities;


import com.tencheeduard.hotelapp.embeddables.RoomId;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
public class Room {

    @EmbeddedId
    RoomId id;

    Integer type;

    Float price;

    Boolean isAvailable;

}

File: frontend/src/app/classes/entities/Room.ts
import { RoomId } from "../ids/RoomId";

export class Room
{
    id : RoomId
    type : number
    price : number
    isAvailable : boolean

    constructor(id : RoomId, type: number, price : number, isAvailable : boolean)
    {
        this.id = id;
        this.type = type;
        this.price = price;
        this.isAvailable = isAvailable;
    }
}